catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/FlashBrickTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.bricks.FlashBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.hardware.SensorTestArduinoServerConnection;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.createProjectAndGetStartScript;
+
+@RunWith(AndroidJUnit4.class)
+public class FlashBrickTest {
+	private int flashBrickPosition;
+	private int waitingTime = 2500;
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		flashBrickPosition = 1;
+		createProjectAndGetStartScript("flashBrickTest").addBrick(new FlashBrick());
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testFlashBrick() {
```

Comment:
ive split the IDE ui tests and stage tests in separate packages.
so this testFlashBrick would go in app, while all other tests would go in separate classes in stage.

this enables you to directly open stage via testRule, after setting parameters in project, which speeds up those tests significantly. (and also makes them more stable since theres no activity switch in the test)

Reply:
Ok thx, the tests are separated now